permissions/permissions-cg.https.html

```diff
@@ -7,20 +7,44 @@
 <script src="/resources/testdriver-vendor.js"></script>
 
 <script>
+async function garbageCollect() {
+  if (typeof TestUtils !== 'undefined' && TestUtils.gc) {
+    await TestUtils.gc();
+  } else if (self.gc) {
+    await self.gc();
+  } else if (self.GCController) {
+    // Present in some WebKit development environments
+    await GCController.collect();
+  } else {
+    var gcRec = function (n) {
+      if (n < 1)
+        return {};
+      var temp = {i: "ab" + i + (i / 100000)};
+      temp += "foo";
+      gcRec(n-1);
+    };
+    for (var i = 0; i < 1000; i++)
+      gcRec(10);
+  }
+}
+
 promise_test(async () => {
   const { state: initialState } = await navigator.permissions.query({
     name: "geolocation",
   });
 
   const pass = new Promise(async (resolve) => {
-    const status = await navigator.permissions.query({
+    let status = await navigator.permissions.query({
       name: "geolocation",
     });
     status.addEventListener("change", resolve, { once: true });
-  });
 
-  // Will cause change event to be dispatched.
-  await test_driver.set_permission({ name: "geolocation" }, "granted");
+    status = null;
+    garbageCollect();
+
+    // Will cause change event to be dispatched.
+    test_driver.set_permission({ name: "geolocation" }, "granted");
```

Comment:
```suggestion
    await test_driver.set_permission({ name: "geolocation" }, "granted");
```

Reply:
Why though? There is no reason we need to wait for this step.

Reply:
it's just in case in the future someone adds more stuff after this step and they forget to await it. Seen it happen too many times.